MYB-386 MYB-388 | Edit favorite

lisandro opened the pull request:
[![MYB388.gif](https://s3.postimg.org/lmzgytx0z/MYB388.gif)](https://postimg.org/image/nesftqgdr/)

lisandro committed:
Check if fav to add already exist. Do not add if duplicated

lisandro committed:
Updated IBOutlets bus rates cell

lisandro committed:
Allowed edit favorite from cell

lisandro committed:
Used root view of app to display progress notification

lisandro committed:
Added favourite cell class for search table

lisandro committed:
Handled errors when user edits a cell

marcosjvivar commented on MyBus/MyBus/Cells/SearchFavoriteTableViewCell.swift:
```diff
@@ -0,0 +1,21 @@
+
+    @IBOutlet weak var favoriteNameLabel: UILabel!
+    @IBOutlet weak var favoriteAddressLabel: UILabel!
+    func loadItem(favorite: RoutePoint) {
```
Can we make an upgrade on this method?

```
func loadItem(favorite: RoutePoint)
{
self.favorite = favorite
setupFields()
}

func setupFields()
{
self.favoriteNameLabel.text = self.favorite.name
self.favoriteAddressLabel.text = self.favorite.address
}
```

marcosjvivar commented on MyBus/MyBus/FavoriteTableViewCell.swift:
```diff
@@ -8,17 +8,20 @@
-    func loadItem(name: String, address: String) {
-        self.name.text = name
-        self.address.text = address
+    func loadItem(favorite: RoutePoint) {
```
Idem _SearchFavoriteTableViewCell_.

lisandro committed:
Extracted setUpFields. Update coordinates of fav address changed

lisandro committed:
Commented start offline pack download

To prevent user from download offline pack map without his authorization using cellular data plan

marcosmvivar commented:
Changes are OK and ready to be merged! 🚀 and :shipit:
